src/test/kotlin/step2/StringCalculatorTest.kt

```diff
@@ -0,0 +1,129 @@
+package step2
+
+import io.kotest.assertions.throwables.shouldNotThrowAny
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.matchers.shouldBe
+
+class StringCalculatorTest : ShouldSpec({
+    context("InputValidation") {
+        should("throw exception if null or empty") {
+            shouldThrow<IllegalArgumentException> {
+                StringCalculator("")
+            }
+        }
+
+        should("throw exception if split strings are less than 3") {
+            listOf("12 34", "12 +", "+ +").forEach {
```

Comment:
How about adding a value to `listOf` that causes this test to fail? (e.g., `12 + 1`)
When the test failed, were you able to quickly identify which input caused the failure?

I recommend checking out Kotest's Data-Driven Testing or Inspectors, which can help with this! 

- [Data Driven Testing](https://kotest.io/docs/framework/datatesting/data-driven-testing.html)
- [Inspectors](https://kotest.io/docs/assertions/inspectors.html)